lib/heapwatch.js

```diff
@@ -21,28 +21,31 @@ class HeapWatch {
         this.timeoutHandle = undefined;
         this.gcReportInterval = undefined;
         this.cumulativeGCTimes = Object.assign({}, ZERO_CUMULATIVE_GC_INTERVAL);
+        this.reportStatsHandler = (stats) => {
+                // Report GC timings to statsd (in nanoseconds).
+                const type = this._gcTypeName(stats.gctype);
+                if (type !== 'unknown') {
+                    this.cumulativeGCTimes[this._gcTypeName(stats.gctype)] += stats.pause;
+                }
+        };
+        this._gcStats = null;
+    }
+
+    _gcTypeName(typeID) {
+        switch (typeID) {
+            case 1: return 'minor';
+            case 2: return 'major';
+            case 4: return 'incremental';
+            case 8: return 'weak';
+            case 15: return 'all';
+            default: return 'unknown';
+        }
     }
 
     setGCMonitor() {
         try {
-            const gcStats = require('gc-stats')();
-            const gcTypeName = (typeID) => {
-                switch (typeID) {
-                    case 1: return 'minor';
-                    case 2: return 'major';
-                    case 4: return 'incremental';
-                    case 8: return 'weak';
-                    case 15: return 'all';
-                    default: return 'unknown';
-                }
-            };
-            gcStats.on('stats', (stats) => {
-                // Report GC timings to statsd (in nanoseconds).
-                const type = gcTypeName(stats.gctype);
-                if (type !== 'unknown') {
-                    this.cumulativeGCTimes[gcTypeName(stats.gctype)] += stats.pause;
-                }
-            });
+            this._gcStats = require('gc-stats')();
```

Comment:
This `require` needs to be guarded, as it's not guaranteed that `gc-stats` is available, and we don't want the process to crash entirely if that is the case.

Reply:
I am not sure I follow. The require is wrapped in a try/catch.

Reply:
Oops, my bad. I don't know how I missed it.